Lib/unittest/case.py

```diff
@@ -274,7 +274,11 @@ def __enter__(self):
                 v.__warningregistry__ = {}
         self.warnings_manager = warnings.catch_warnings(record=True)
         self.warnings = self.warnings_manager.__enter__()
-        warnings.simplefilter("always", self.expected)
+        if isinstance(self.expected, tuple):
```

Comment:
This should probably be done in a separate PR, but `issubclass` does not support a tuple. However, the type of the `category` parameter of `warnings.simplefilter` doesn't really indicate that it should support a tuple anyway.
I could also change it so that the `issubclass` checks for `tuples` before doing the check, but I'm not sure if that is actually correct.

Let me know!

Reply:
I think that there may be a third-party code which uses `simplefilter()` with a tuple. Instead of changing unittest, it is better to make the argument check in `simplefilter()` more lenient.

It would be nice to make also `filterwarnings()` accepting a tuple.